R/helpers_data.R

```diff
@@ -0,0 +1,235 @@
+    df$dose[df$dose == pbo_dose] <- 0
+  }
+  df
+}
+
+# h_plot_data_df ----
+
+## generic ----
+
+#' Helper Function for the Plot Method of subclasses of [`GeneralData`]
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' A method that transforms [`GeneralData`]  objects into a `tibble` suitable for
+#' plotting with `ggplot2` methods
+#'
+#' @param data (`GeneralData`)\cr object from which data is extracted and converted
+#' into a data frame.
+#' @param ... further arguments passed to class-specific methods.
+#' @return `data.frame` containing columns for patient, cohort, dose and toxicity grade
+#' @aliases h_plot_data_df
+#'
+setGeneric(
+  name = "h_plot_data_df",
+  def = function(data, ...) standardGeneric("h_plot_data_df"),
+  valueClass = "data.frame"
+)
+
+# Data ----
+
+#' Helper Function for the Plot Method of [`Data`]
+#'
+#' @param data (`Data`)\cr object from which data is extracted and converted
+#'   into a data frame.
+#' @param blind (`flag`)\cr should data be blinded?
+#'   If `TRUE`, then for each cohort, all DLTs are assigned to the first
+#'   subjects in the cohort. In addition, the placebo (if any) is set to the
+#'   active dose level for that cohort.
+#' @param legend (`flag`)\cr Display the legend for the toxicity categories
+#' @param ... further arguments passed to `data.frame` constructor.
+#'   It can be e.g. an extra `column_name = value` pair based on a slot
+#'   from `x` (which in this case might be a subclass of `Data`)
+#'   which does not appear in `Data`.
+#' @return A [`data.frame`] object with columns patient, ID, cohort, dose and
+#' toxicity.
+#' @describeIn h_plot_data_df Class specific method for [`Data`]
+setMethod(
+  f = "h_plot_data_df",
+  signature = signature(data = "Data"),
+  definition = function(data, blind = FALSE, legend = TRUE, ...) {
+    df <- data.frame(
+      patient = seq_along(data@x),
+      ID = paste(" ", data@ID),
+      cohort = data@cohort,
+      dose = data@x,
+      toxicity = ifelse(data@y == 1, "Yes", "No"),
+      ...
+    )
+    df <- h_blind_plot_data(df, blind, data@placebo, data@doseGrid[1])
+    df
+  }
+)
+
+# DataOrdinal ----
+
+#' Helper Function for the Plot Method of [`DataOrdinal`]
+#'
+#' @param data (`DataOrdinal`)\cr object from which data is extracted and converted
+#'   into a data frame.
+#' @param blind (`flag`)\cr should data be blinded?
+#'   If `TRUE`, then for each cohort, all DLTs are assigned to the first
+#'   subjects in the cohort. In addition, the placebo (if any) is set to the
+#'   active dose level for that cohort.
+#' @param ... further arguments passed to `data.frame` constructor.
+#'   It can be e.g. an extra `column_name = value` pair based on a slot
+#'   from `x` (which in this case might be a subclass of `Data`)
+#'   which does not appear in `Data`.
+#' @return A [`data.frame`] object with values to plot.
+#' @describeIn h_plot_data_df Class specific method for [`DataOrdinal`]
```

Comment:
when we describe this in the same page then we don't need to / should not repeat the argument description (otherwise it will be overwritten)

Reply:
What change are you requesting here?

Reply:
Because we use `@describeIn`, any arguments or return values that are documented in multiple of the `@describeIn` parts will be overwritten. So we should remove the `@param` and `@return` parts here because they are already documented in the parent generic function roxygen chunk